src/gmpy2.c

```diff
@@ -996,21 +996,12 @@ PyMODINIT_FUNC PyInit_gmpy2(void)
 
     /* Validate the sizes of the various typedef'ed integer types. */
 
-#if defined _WIN64 && MPIR
-    if (sizeof(mp_bitcnt_t) != sizeof(PY_LONG_LONG)) {
-        /* LCOV_EXCL_START */
-        SYSTEM_ERROR("Size of PY_LONG_LONG and mp_bitcnt_t not compatible (_WIN64 && MPIR)");
-        return NULL;;
-        /* LCOV_EXCL_STOP */
-    }
-#else
```

Comment:
This should be kept as it is becausse `sizeof(mp_bitcnt_t) == 64` whereas `sizeof(long) == 32`

Reply:
I would like to change this to:

#if defined _WIN64 && HAVE_64BIT_BITCNT

I've experimented with a custom Windows GMP build that defines a 64-bit mp_bitcnt_t. I'd like to detect that scenario in a  neutral manner. This will simplify making the appropriate #define's in gmpy2_convert_utils.h.

@isuruf Can you add that define to the conda build process?

Case

Reply:
I could add that, but wouldn't it be better to just check `sizeof(mp_bitcnt_t) == 8` instead of having a macro `HAVE_64BIT_BITCNT`?

Reply:
Hmm. I think I can do that.

I'll write a GMPy_Integer_AsMpBitCnt function that checks sizeof(mp_bitcnt_t) and calls the appropriate Long versus LongLong function. That will eliminate the hacks in gmpy2_convert_utils.h

@skirpichev Please remove all the checks of sizeof(mp_bitcnt_t) in gmpy2.c. They will no longer be needed.

Reply:
> Please remove all the checks of sizeof(mp_bitcnt_t) in gmpy2.c. They will no longer be needed.

Done, see [2d85f34](https://github.com/aleaxit/gmpy/pull/395/commits/2d85f34619f99bc79a2a4e92ee4afe6f03393878)

> I'll write a GMPy_Integer_AsMpBitCnt function that checks sizeof(mp_bitcnt_t) and calls the appropriate Long versus LongLong function.

See the last commit.